### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by ForTehLose**
lots of clean up to enabled cargo fmt, cargo clippy, and cargo rustdoc

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -162,7 +162,8 @@ impl Plugin for OpenXrPlugin {
             let mut manual_texture_views = app.world.resource_mut::<ManualTextureViews>();
             manual_texture_views.insert(LEFT_XR_TEXTURE_HANDLE, left);
             manual_texture_views.insert(RIGHT_XR_TEXTURE_HANDLE, right);
-            drop(manual_texture_views);
+            //drop non drop
+            //drop(manual_texture_views);
```
Comment (ForTehLose) [inline src/lib.rs:166]: this was a change I wasn't sure of

File: src/lib.rs [hunk 2]
```diff
@@ -31,7 +31,7 @@ pub const RIGHT_XR_TEXTURE_HANDLE: ManualTextureViewHandle = ManualTextureViewHa
 /// Adds OpenXR support to an App
 #[derive(Default)]
 pub struct OpenXrPlugin;
-
+#[allow(clippy::type_complexity)]
```
Comment (ForTehLose) [inline src/lib.rs:34]: I think this could be simplified, but lacked the expertise to be sure

**Comment by oli-obk** [comment]
This PR needs a rebase

File: src/lib.rs (hunk 1 above)
Comment (oli-obk) [reply src/lib.rs:166]: yea this seems fine to just remove. Probably leftover from an older bevy version where this was `Drop`

File: src/graphics/mod.rs [hunk 3]
```diff
@@ -12,6 +12,7 @@ use crate::resources::{
 
 use openxr as xr;
 
+#[allow(clippy::type_complexity)]
```
Comment (oli-obk) [inline src/graphics/mod.rs:15]: could add this to cargo.toml (latest Rust version has a section for it):

```toml
[lints.clippy]
type_complexity = "allow"
too_many_arguments = "allow"
```

**Comment by MalekiRe** [comment]
Stale

**Pull request closed without merging**